src/arch/aarch64/mod.rs

```diff
@@ -301,8 +303,12 @@ pub fn init_guest_mem(
 		if idx_l1 == 0 && idx_l2 == 0 && idx_l3 == 0 && idx_l4 == 0 {
 			// Hypercall/IO mapping
 			pmd_slice[idx_l1] = frame_addr | PT_MEM_CD;
-		} else {
+		} else if idx_l1 == 0 && idx_l2 == 0 && idx_l3 == 0 && idx_l4 < 4 {
```

Comment:
Are you sure, that this check is sufficient to ensure the required alignment of this flag?
Arm v8 reference Manual states:
> The contiguous output address range must be aligned to size of [16/32/128] translation table entries [depending on the page granule] at the same translation table level.

I don't recall which granule we are using here, but we need an alignment of at least 16 entries.

Reply:
You are right. A small typo... In case 4 KBytes pages, the range must be aligned to size of 16 translation table entries.